js/state/siteUtil.js

```diff
@@ -417,6 +417,10 @@ module.exports.removeSite = function (state, siteDetail, tag, reorder = true, sy
     site = site.set('tags', tags)
     return state.setIn(stateKey, site)
   } else {
+    if (!tag && module.exports.isHistoryEntry(siteDetail)) {
+      // Delete the site from history
+      return state.deleteIn(stateKey)
```

Comment:
@darkdh `module.exports.isHistoryEntry` verifies that the site is not a bookmark or a pinned tab before deleting it. but you're right that #10136 probably affects this PR